Riot/Modules/Room/RoomCoordinator.swift

```diff
@@ -246,6 +254,47 @@ final class RoomCoordinator: NSObject, RoomCoordinatorProtocol {
         }
     }
     
+    private func loadRoom(withUserId userId: String) {
+        // Start a new discussion
+            
+        // Present activity indicator when retrieving roomDataSource for given room ID
+        startLoading()
+        
+        // Try to search target user if not exist in local session
+        if let user = self.parameters.session.getOrCreateUser(userId), user.displayname != nil {
+            // User has already been found from local session no update needed
+            self.stopLoading()
+            
+            // Update RoomViewController with found target user
+            self.roomViewController.displayNewDirectChat(withTargetUser: user, session: self.parameters.session)
+        } else if let user = MXUser(userId: userId) {
+            user.update(fromHomeserverOfMatrixSession: self.parameters.session) {
+                self.stopLoading()
+                
+                self.parameters.session.store.store(user)
+                
+                // Update RoomViewController with found target user
+                self.roomViewController.displayNewDirectChat(withTargetUser: user, session: self.parameters.session)
+            } failure: { [weak self] error in
+                guard let self = self else { return }
+                self.stopLoading()
+                
+                MXLog.error("[RoomCoordinator] User does not exist")
+                
+                // Alert user
+                let alert = UIAlertController(title: nil, message: VectorL10n.roomCreationDmError, preferredStyle: .alert)
+                
+                let cancelActionTitle = VectorL10n.ok
+                let cancelAction = UIAlertAction(title: cancelActionTitle, style: .cancel) { (_) in
+                    self.roomViewController.popToHomeViewController()
```

Comment:
This is not a good pattern for a coordinator. Instead we should callback a delegate method and delegate should decide where to go.

Reply:
Yes, I totally agree with that. I missed it in my first review of this PR.
I fixed it in the last commit.